### PR TITLE
Add Pax singleprocess train and eval tests

### PR DESCRIPTION
**Pull request by ashors1**
Adds the following singleprocess tests to the Pax test suite: 

- eval (upstream pax)
- eval + TE (rosetta pax)
- 8-way DP (upstream pax)
- 8-way DP + TE (rosetta pax)
- 2 FSDP + 4 TP (upstream pax)
- 2 FSDP + 4 TP + TE (rosetta pax)

### CODE REVIEW

**Comment by ashors1** [comment]
Upstream Pax workflow: [https://github.com/NVIDIA/JAX-Toolbox/actions/runs/6702046559](https://github.com/NVIDIA/JAX-Toolbox/actions/runs/6702046559)
Rosetta workflow: [https://github.com/NVIDIA/JAX-Toolbox/actions/runs/6702059749](https://github.com/NVIDIA/JAX-Toolbox/actions/runs/6702059749).

8-way DP singleprocess tests are failing in upstream and Rosetta with a known mutex error:
```scipy.stats._statlib[mutex.cc : 1954] RAW: Check (v & (kMuWriter | kMuReader)) != (kMuWriter | kMuReader) failed: Lock: Mutex corrupt: both reader and writer lock held: 0xffffffffffffffff```

The 2 FSDP + 4 TP + TE singleprocess test also fails nondeterministically with a similar error.

File: .github/container/test-pax.sh [hunk 1]
```diff
@@ -306,19 +317,45 @@ else:
...
+      task_p.model.report_strict_acc=True
+
       return task_p
 
 EOF
 
 ## Launch
 set -ex
-ENABLE_TE=$ENABLE_TE python -m paxml.main \
+if [[ ${EVALUATE} -ne 0 ]]; then
```
Comment (terrykong) [inline .github/container/test-pax.sh:329]: nit: can save characters by just exporting ENABLE_TE
```suggestion
export ENABLE_TE=${ENABLE_TE}
if [[ ${EVALUATE} -ne 0 ]]; then
```
and removing it from each python call

**Comment by terrykong** [comment]
Other than small comment. LGTM

**Review by terrykong — APPROVED** [review]

**Review by yhtang — COMMENTED** [review]
Shall we hold off merging this until the existing issues all got fixed?

**Comment by ashors1** [comment]
Updated upstream Pax and Rosetta Pax tests: 
- [https://github.com/NVIDIA/JAX-Toolbox/actions/runs/6853025145](https://github.com/NVIDIA/JAX-Toolbox/actions/runs/6853025145)
- [https://github.com/NVIDIA/JAX-Toolbox/actions/runs/6853026684](https://github.com/NVIDIA/JAX-Toolbox/actions/runs/6853026684)

**Comment by terrykong** [comment]
LGTM. Only error appears to be the known TP+DP+PP failure. FFTM

**Pull request merged**